.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: Main
+
+on:
+  push:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  Build:
+    name: Build
+    permissions:
+      statuses: write
+      pull-requests: write
+    uses: ./.github/workflows/_build.yml
+
+  StorybookDeploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set the correct npm version
+        run: nvm install && nvm use
+
+      - name: Install dependencies from npm
+        run: |
+          nvm use
+          npm ci
+
+      - name: Build Backpack
+        run: |
+          nvm use
+          npm run build
+          npm run storybook:dist
+      
+      - name: Deploy Storybook for main
+        uses: peaceiris/actions-gh-pages@v3
+        if: github.ref == 'refs/heads/main'
+        with:
+          personal_token: ${{ secrets.DEPLOY_TOKEN }}
+          publish_dir: dist-storybook/
+          keep_files: false
+          external_repository: backpack/storybook
+          publish_branch: main
```

Comment:
For now we will duplicate some steps in the process just to get it to a state for automation and we have a follow up ticket as part of this epic to tidy and improve the workflow later on
So will look to solve cleaner and faster CI with caching at that point